process/accounts/accountsProcessor.go

```diff
@@ -11,53 +11,39 @@ import (
 	"github.com/ElrondNetwork/elastic-indexer-go/data"
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
-	coreData "github.com/ElrondNetwork/elrond-go-core/data"
 	"github.com/ElrondNetwork/elrond-go-core/data/esdt"
-	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	coreIndexerData "github.com/ElrondNetwork/elrond-go-core/data/indexer"
 	logger "github.com/ElrondNetwork/elrond-go-logger"
-	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 )
 
 var log = logger.GetOrCreate("indexer/process/accounts")
 
 // accountsProcessor a is structure responsible for processing accounts
 type accountsProcessor struct {
-	internalMarshalizer    marshal.Marshalizer
 	addressPubkeyConverter core.PubkeyConverter
-	accountsDB             indexer.AccountsAdapter
 	balanceConverter       indexer.BalanceConverter
 }
 
 // NewAccountsProcessor will create a new instance of accounts processor
 func NewAccountsProcessor(
-	marshalizer marshal.Marshalizer,
 	addressPubkeyConverter core.PubkeyConverter,
-	accountsDB indexer.AccountsAdapter,
 	balanceConverter indexer.BalanceConverter,
 ) (*accountsProcessor, error) {
-	if check.IfNil(marshalizer) {
-		return nil, indexer.ErrNilMarshalizer
-	}
 	if check.IfNil(addressPubkeyConverter) {
 		return nil, indexer.ErrNilPubkeyConverter
 	}
-	if check.IfNil(accountsDB) {
-		return nil, indexer.ErrNilAccountsDB
-	}
 	if check.IfNil(balanceConverter) {
 		return nil, indexer.ErrNilBalanceConverter
 	}
 
 	return &accountsProcessor{
-		internalMarshalizer:    marshalizer,
 		addressPubkeyConverter: addressPubkeyConverter,
-		accountsDB:             accountsDB,
 		balanceConverter:       balanceConverter,
 	}, nil
 }
 
 // GetAccounts will get accounts for regular operations and esdt operations
-func (ap *accountsProcessor) GetAccounts(alteredAccounts data.AlteredAccountsHandler) ([]*data.Account, []*data.AccountESDT) {
+func (ap *accountsProcessor) GetAccounts(alteredAccounts data.AlteredAccountsHandler, coreAlteredAccounts map[string]*coreIndexerData.AlteredAccount) ([]*data.Account, []*data.AccountESDT) {
```

Comment:
Here in the `accountsProcessor` you can add a TODO (and a JIRA task)  in order to remove all the parts with the `alteredAccounts` because you already to this part at the protocol level.

Reply:
added todo + task EN-12389